WinUIGallery/ControlPages/DesignGuidance/ColorsPage.xaml.cs

```diff
@@ -12,41 +11,34 @@ public sealed partial class ColorsPage : Page
         public ColorsPage()
         {
             this.InitializeComponent();
-            NavigationFrame.Navigate(typeof(TextSection));
         }
-
-        private void ToggleButton_Checked(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
+        private void OnSelectionChanged(object sender, SelectionChangedEventArgs e)
         {
-            if(sender is ToggleButton item)
+            switch (PageSelector.SelectedIndex)
             {
-                switch (item.Content)
-                {
-                    case "Text":
-                        NavigationFrame.Navigate(typeof(TextSection));
-                        break;
-                    case "Fill":
-                        NavigationFrame.Navigate(typeof(FillSection));
-                        break;
-                    case "Stroke":
-                        NavigationFrame.Navigate(typeof(StrokeSection));
-                        break;
-                    case "Background":
-                        NavigationFrame.Navigate(typeof(BackgroundSection));
-                        break;
-                    case "Signal":
-                        NavigationFrame.Navigate(typeof(SignalSection));
-                        break;
-                }
+                case 0:
+                    NavigationFrame.Navigate(typeof(TextSection));
+                    break;
+                case 1:
+                    NavigationFrame.Navigate(typeof(FillSection));
+                    break;
+                case 2:
+                    NavigationFrame.Navigate(typeof(StrokeSection));
+                    break;
+                case 3:
+                    NavigationFrame.Navigate(typeof(BackgroundSection));
+                    break;
+                case 4:
+                    NavigationFrame.Navigate(typeof(SignalSection));
+                    break;
             }
-            foreach(var child in SelectionElement.Children)
-            {
-                if (child != sender)
-                {
-                    if (child is ToggleButton button)
+        }
 
-                        button.IsChecked = false;
-                }
-            }
+        private void Page_Loaded(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
+        {
+            PageSelector.SelectionChanged -= OnSelectionChanged;
+            NavigationFrame.Navigate(typeof(TextSection));
+            PageSelector.SelectionChanged += OnSelectionChanged;
```

Comment:
Why is this needed? Especially hooking up the SelectionChanged event?

Reply:
Updated, a bit cleaner now :)